### DISCUSSION TITLE
Object.lockUniScaling is removed. Alternatives to get the same identical functionality with less code are being evaluated. breaking:

### DISCUSSION THREAD

**Discussion by TimLiu1** [fabricjs/fabric.js#7402] [Q&A]
which function I can do same thing?

**Comment by melchiar** [comment]
`Canvas.uniformScaling`

see http://fabricjs.com/v4-breaking-changes

**Comment by asturur** [comment]
Which is the `same thing`? what function are you after?

**Comment by basielsmitz** [comment]
I need this property as well. I work with certain objects that need to keep their proportions while scaling without having the user press the `uniScaleKey`, because it shouldn't be optional for those objects. 

Previously when i enabled `lockUniScaling` i would only get the diagonal akerpoints and the object would keep its proportions while scaling. 

I can hide the unecessary control points with `setControlsVisibility` but the diagonal points still scale width and height independently from eachother. I'm still in the process of finding a solution. 

The only idea I currently have is to listen to the `object:scaling` event for those certain objects and check if `__corner` is bl, br, tl or tr then set `Canvas.uniformScaling` to true for the duration of the scaling and finally reset it afterwards. 

Although I would prefere to be able to just enable a property on the objects independently

**Comment by basielsmitz** [comment]
@TimLiu1 

this is my current workaround


```
fabricObject.setControlsVisibility({
    mb: false,
    ml: false,
    mr: false,
    mt: false
});

let timer;

fabricObject.on('scaling', e => {
  if (['bl', 'br', 'tr', 'tl'].indexOf(e.transform.corner) !== -1) {
    clearTimeout(timer);
    canvas.uniformScaling = true;
    timer = setTimeout(() => {canvas.uniformScaling = false}, 500);
  }
});
```

**Comment by TimLiu1** [comment]
> @TimLiu1
> 
> this is my current workaround
> 
> ```
> fabricObject.setControlsVisibility({
>     mb: false,
>     ml: false,
>     mr: false,
>     mt: false
> });
> 
> let timer;
> 
> fabricObject.on('scaling', e => {
>   if (['bl', 'br', 'tr', 'tl'].indexOf(e.transform.corner) !== -0) {
>     clearTimeout(timer);
>     canvas.uniformScaling = true;
>     timer = setTimeout(() => {canvas.uniformScaling = false}, 500);
>   }
> });
> ```

thank you very much

**Comment by TimLiu1** [comment]
> Which is the `same thing`? what function are you after?

by control  canvas.uniformScaling = true/false to resolve

**Comment by TimLiu1** [comment]
> `Canvas.uniformScaling`
> 
> see http://fabricjs.com/v4-breaking-changes

I think object uniformScaling overwrite canvas uniformScaling is make sense

**Comment by DineshCodeFlow** [comment]
This worked for me
```
            this.canvas.on('object:scaling', e => {
                if (['bl', 'br', 'tr', 'tl'].indexOf(e.transform.corner) !== -1) {
                canvas.uniformScaling = false;
                }
            });
```
